src/CIFAR10/CIFAR10.jl

```diff
@@ -0,0 +1,95 @@
+export CIFAR10
+module CIFAR10
+    using DataDeps
+    using BinDeps
+    using ImageCore
+    using ColorTypes
+    using FixedPointNumbers
+    using ..bytes_to_type
+    using ..datafile
+    using ..download_dep
+    using ..download_docstring
+
+    export
+
+        classnames,
+
+        traintensor,
+        testtensor,
+
+        trainlabels,
+        testlabels,
+
+        traindata,
+        testdata,
+
+        convert2image,
+        convert2features,
+
+        download
+
+    const DEPNAME = "CIFAR10"
+    const NCHUNKS = 5
+
+    filename_for_chunk(file_index::Int) =
+        joinpath("cifar-10-batches-bin", "data_batch_$(file_index).bin")
+
+    const TESTSET_FILENAME =
+        joinpath("cifar-10-batches-bin", "test_batch.bin")
+
+    const CLASSES = [
+        "airplane",
+        "automobile",
+        "bird",
+        "cat",
+        "deer",
+        "dog",
+        "frog",
+        "horse",
+        "ship",
+        "truck",
+    ]
+
+    download(args...; kw...) = download_dep(DEPNAME, args...; kw...)
+
+    include(joinpath("Reader","Reader.jl"))
+    include("interface.jl")
+    include("utils.jl")
+
+    function __init__()
+        RegisterDataDep(
+            DEPNAME,
+            """
+            Dataset: The CIFAR-10 dataset
+            Authors: Alex Krizhevsky, Vinod Nair, Geoffrey Hinton
+            Website: https://www.cs.toronto.edu/~kriz/cifar.html
+            Reference: https://www.cs.toronto.edu/~kriz/learning-features-2009-TR.pdf
+
+            [Krizhevsky, 2009]
+                Alex Krizhevsky.
+                "Learning Multiple Layers of Features from Tiny Images",
+                Tech Report, 2009.
+
+            The CIFAR-10 dataset is a labeled subsets of the 80
+            million tiny images dataset. It consists of 60000
+            32x32 colour images in 10 classes, with 6000 images
+            per class.
+
+            The compressed archive file that contains the
+            complete dataset is available for download at the
+            offical website linked above; specifically the binary
+            version for C programs. We can download and unpack
+            this archive for you if you wish, but that doesn't
+            free you from the burden of using the data
+            responsibly and respect copyright. The authors of
+            CIFAR-10 aren't really explicit about any terms of
+            use, so please read the website to make sure you want
+            to download the dataset.
+            """,
+            "https://www.cs.toronto.edu/~kriz/cifar-10-binary.tar.gz",
+            "c4a38c50a1bc5f3a1c5537f2155ab9d68f9f25eb1ed8d9ddda3db29a59bca1dd",
+            fetch_method = (src, dst) -> run(BinDeps.download_cmd(src, dst)),
```

Comment:
Why `BinDeps.download_cmd` ?
and not `Base.download` ? (which is the default if not specified)

Compare and contrast:

 - https://github.com/JuliaLang/julia/blob/195008690739cc98719773c734a07b41a618906c/base/interactiveutil.jl#L640-L692
 - https://github.com/JuliaLang/BinDeps.jl/blob/803ca7975d8b5f43610fcb7504f09f2e573b82fd/src/BinDeps.jl#L51-L84

Reply:
unsure, somehow I expected the BinDeps one to be more general/cross platform. And that impression was solely based on the fact that it exists (why else does it exist?)

Reply:
I have wondered why it existed since I first saw it.
I think it might be justreally old, older than `Base.download` (which is itself old).

It might be more general, but it is redundantly so.
In the julia already promises `Base.download` will work-on every platform.
(Or so I believe)

Reply:
Well I don't feel strongly about this, so I'd be ok to change it to download (or delete the line, since download is the default anyway)